_docs/tasks/security/https-overlay.md

```diff
@@ -0,0 +1,214 @@
+
+### Generate certificates
+
+You need to have openssl installed to run this command
+
+```bash
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
+```
+
+```bash
+$ kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt
+secret "nginxsecret" created
+```
+
+### Create an https service without Istio sidecar
+
+This subsection creates a nginx-based https service.
+
+First we need to create a configmap:
+
+```bash
+$ kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf
+configmap "nginxconfigmap" created
+```
+
+Then deploy the service:
+
+```bash
+$ kubectl apply -f samples/https/nginx-app.yaml
+...
+service "nginxsvc" created
+replicationcontroller "my-nginx" created
+```
+
+Then, create another pod to call this service.
+
+```bash
+$ kubectl apply -f <(bin/istioctl kube-inject --debug -f samples/sleep/sleep.yaml)
+```
+
+Get the pods
+
+```bash
+$ kubectl get pod
+NAME                              READY     STATUS    RESTARTS   AGE
+my-nginx-jwwck                    2/2       Running   0          1h
+sleep-847544bbfc-d27jg            2/2       Running   0          18h
+```
+
+Ssh into the istio-proxy container of sleep pod.
+```bash
+$ kubectl exec -it sleep-847544bbfc-d27jg -c istio-proxy /bin/bash
+```
+
+Call nginxsvc
+```bash
+# curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+You can actually combine the above two command into one:
+
+```bash
+$ kubectl exec sleep-847544bbfc-d27jg -c istio-proxy -- curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+### Create an https service with Istio sidecar with mTLS disabled
+
+In "Before you begin" section, the istio control plane is deployed with mTLS
+disabled. So you only need to redeploy the nginx https service with sidecar.
+
+Delete the https service.
+```bash
+$ kubectl delete -f nginx-app.yaml
+```
+
+Deploy it with sidecar
+
+```bash
+$ kubectl apply -f <(bin/istioctl kube-inject --debug -f samples/https/nginx-app.yaml)
+```
+
+Make sure the pod is up and running
+
+```bash
+$ kubectl get pod
+NAME                              READY     STATUS    RESTARTS   AGE
+my-nginx-6svcc                    2/2       Running   0          1h
+sleep-847544bbfc-d27jg            2/2       Running   0          18h
+```
+
+And run
+```bash
+$ kubectl exec sleep-847544bbfc-d27jg -c sleep -- curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+If you run from istio-proxy container, it should work as well
+```bash
+$ kubectl exec sleep-847544bbfc-d27jg -c istio-proxy -- curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+Note: this example is borrowed from [kubernetes examples](https://github.com/kubernetes/examples/blob/master/staging/https-nginx/README.md).
+
+### Create an https service with Istio sidecar with mTLS enabled
+
+You need to redeploy the Istio control plane with mTLS enabled.
```

Comment:
Or you can also refer to per-service enable/disable mTLS option :). I'll add using authn policy for that soon, and it would be easier.

Reply:
Alternatively, you can assume/state the instruction start from scratch (istio is not install, apps are not install), and give a hint what to do if they want to switch. It would be easier for user to jump to this demo right away, instead of following the previous ones.

Reply:
Good point, updated.